### PR TITLE
Locating Nearest Facility with Origin-Destination Matrix

### PR DESCRIPTION
**Pull request by Santhosh-M31**
…est Facility with Origin-Destination Matrix

### CODE REVIEW

File: source/docs/3/origin_destination_matrix.rst [hunk 1]
```diff
@@ -60,27 +60,31 @@ Procedure
   .. image:: /static/3/origin_destination_matrix/images/2.png
```
Comment (spatialthoughts) [inline source/docs/3/origin_destination_matrix.rst:60]: Fix step 2. Not using wards.

File: source/docs/3/origin_destination_matrix.rst [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 
 .. note::
 
-  This tutorial shows how to use your own network data to compute an origin-destination matrix. If you do not have your own network data, you can use **ORS Tools Plugin** and algorithm :menuselection:`ORS Tools --> Matrix -->  Matrix from Layers` to do the similar analysis using OpenStreetMap data. See :doc:`service_area_analysis` to learn how to use ORS Tools plugin.
+  This tutorial shows how to use your own network data to compute an origin-destination matrix. If you do not have your own network data, you can use **ORS Tools Plugin** and algorithm :menuselection:`ORS Tools --> Matrix -->  Matrix from Layers` to do a similar analysis using OpenStreetMap data. See :doc:`service_area_analysis` to learn how to use ORS Tools plugin.
 
 Overview of the task
```
Comment (spatialthoughts) [inline source/docs/3/origin_destination_matrix.rst:10]: Other skills, remove 'stratified sample' nad making it 'random sample'

File: source/docs/3/origin_destination_matrix.rst [hunk 3]
```diff
@@ -90,115 +94,57 @@ Procedure
   .. image:: /static/3/origin_destination_matrix/images/8.png
     :align: center
   
-9. This algorithm helps find the distances along the network between selected origin and destination layers. Select ``Street_Centerlines`` as the :guilabel:`Network layer`. Select ``origin_points`` as the :guilabel:`From-Points layer` and ``OBJECTID`` as the :guilabel:`Unique Point ID field`. Similarly, set ``destination_points`` as the :guilabel:`To-Points Layer` and ``OBJECTID`` as  the :guilabel:`Unique Point ID field`. Set the :guilabel:`Optimization Criterion` as ``Shortest Path``.
+9. This algorithm helps find the distances along with the network between selected origin and destination layers. Select ``Roadway_Block`` as the :guilabel:`Network layer`. Select ``origin_points`` as the :guilabel:`From-Points layer` and ``OBJECTID`` as the :guilabel:`Unique Point ID field`. Similarly, set ``destination_points`` as the :guilabel:`To-Points Layer` and ``OBJECTID`` as  the :guilabel:`Unique Point ID field`. Set the :guilabel:`Optimization Criterion` as ``Shortest Path (distance optimization)``.
```
Comment (spatialthoughts) [inline source/docs/3/origin_destination_matrix.rst:97]: Need to be OBJECTID_1, fix both text , screenshot and subsequent resulrs.

File: source/docs/3/origin_destination_matrix.rst [hunk 4]
```diff
@@ -90,115 +94,57 @@ Procedure
...
+13. Search for and locate the :menuselection:`Vector general --> Execute SQL`, select ``...`` in :guilabel:`Additional input data sources` check the :guilabel:`Output OD Matrix` and, click :guilabel:`OK`. Now click the :guilabel:`Summation` under :guilabel:`SQL query`.
 
   .. image:: /static/3/origin_destination_matrix/images/13.png
     :align: center
   
-14. Enter the following query and click :guilabel:`Execute`. The results will be displayed in the panel below. As expected, we have 9 rows in the result - the shortest path destination for each origin point. Check and select :guilabel:`Column with unique values` as ``origin_id``. Enter ``nearest_destinations`` as the :guilabel:`Layer name (prefix)`. Click :guilabel:`Load`.
+14. Enter the following query in :guilabel:`SQL query` dialog box. Enter ``geometry`` as the :guilabel:`Geometry field` and, select ``LineString`` as the :guilabel:`Geometry type`. Click :guilabel:`Run`.
```
Comment (spatialthoughts) [inline source/docs/3/origin_destination_matrix.rst:122]: Fix query code as shown in screenshot

File: source/docs/3/origin_destination_matrix.rst [hunk 5]
```diff
@@ -90,115 +94,57 @@ Procedure
...
 
   .. image:: /static/3/origin_destination_matrix/images/15.png
     :align: center
   
-16. Select ``origin_points`` as the :guilabel:`Input layer` and ``OBJECTID`` as the :guilabel:`Table field`. Set ``nearest_destinations`` as the :guilabel:`Input layer 2` and ``origin_id`` as the :guilabel:`Table field 2`. Click the :guilabel:`...` button next to :guilabel:`Layer 2 fields to copy` and select ``destination_id`` and ``shortest_distance``. Click :guilabel:`Run`.
+16. Note that even though the lines connecting the origin and destination is a straight-line, the destination was found using the distance along with the network. It will be much useful visualization to show the actual shortest-path between each origin-destination. As of now, there is no easy way to generate the shortest-path between multiple origin-destination pairs the way we generated the distance matrix. But I will demonstrate a way to use some python scripting to generate this visualization. First, let's run the shortest path algorithm on 1 pair. Locate the :menuselection:`QNEAT3 --> Routing --> Shortest path (point to point)` algorithm and launch it.
```
Comment (spatialthoughts) [inline source/docs/3/origin_destination_matrix.rst:137]: This text from this step onwards doesn't make sense at all. There are important validation steps missing. Please refer to chat screenshot. Remove reference to python scripting. Say we will validate the results using 1 pair.

File: source/docs/3/origin_destination_matrix.rst (hunk 2 above)
Comment (Santhosh-M31) [reply source/docs/3/origin_destination_matrix.rst:10]: Updated.

File: source/docs/3/origin_destination_matrix.rst (hunk 3 above)
Comment (Santhosh-M31) [reply source/docs/3/origin_destination_matrix.rst:97]: updated.

File: source/docs/3/origin_destination_matrix.rst (hunk 4 above)
Comment (Santhosh-M31) [reply source/docs/3/origin_destination_matrix.rst:122]: updated.

File: source/docs/3/origin_destination_matrix.rst (hunk 1 above)
Comment (Santhosh-M31) [reply source/docs/3/origin_destination_matrix.rst:60]: Updated.

File: source/docs/3/origin_destination_matrix.rst (hunk 5 above)
Comment (Santhosh-M31) [reply source/docs/3/origin_destination_matrix.rst:137]: Updated.

**Pull request merged**